### PR TITLE
Fail at configure-time for C++17 with NVCC when relaxed constexpr support not enabled

### PR DESCRIPTION
**Pull request by dalg24**
Avoid failing at build time and confusing users

### CODE REVIEW

**Comment by dalg24** [comment]
Did not think before opening.  This is only an issue downstream if users do not enable relaxed constexpr and use it but there is no reason to require it

File: cmake/kokkos_corner_cases.cmake [hunk 1]
```diff
@@ -49,11 +49,14 @@ ENDIF()
...
+    MESSAGE(FATAL_ERROR "You have requested C++17 support for GCC ${KOKKOS_CXX_COMPILER_VERSION}. Although CMake has allowed this and GCC accepts -std=c++1z/c++17, GCC <= 6 does not properly support *this capture. Please reduce the C++ standard to 14 or upgrade the compiler if you do need C++17 support.")
   ENDIF()
 
   IF (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 11)
-    MESSAGE(FATAL_ERROR "You have requested c++17 support for NVCC ${KOKKOS_CXX_COMPILER_VERSION}. NVCC only supports C++17 from version 11 on. Please reduce the C++ standard to 14 or upgrade the compiler if you need C++17 support.")
+    MESSAGE(FATAL_ERROR "You have requested C++17 support for NVCC ${KOKKOS_CXX_COMPILER_VERSION}. NVCC only supports C++17 from version 11 on. Please reduce the C++ standard to 14 or upgrade the compiler if you need C++17 support.")
+  ENDIF()
+  IF (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND NOT KOKKOS_ENABLE_CUDA_CONSTEXPR)
```
Comment (lkersting) [inline cmake/kokkos_corner_cases.cmake:58]: I think the issue happens when you are trying to use C++17 and you have `Kokkos_ENABLE_CUDA_CONSTEXPR=ON`
```suggestion
  IF (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND KOKKOS_ENABLE_CUDA_CONSTEXPR)
```

File: cmake/kokkos_corner_cases.cmake [hunk 2]
```diff
@@ -49,11 +49,14 @@ ENDIF()
...
+    MESSAGE(FATAL_ERROR "You have requested C++17 support for NVCC ${KOKKOS_CXX_COMPILER_VERSION}, but you forgot to enable relaxed constexpr. Please reduce the C++ standard to 14 or re-configure with -DKokkos_ENABLE_CUDA_CONSTEXPR=ON")
```
Comment (lkersting) [inline cmake/kokkos_corner_cases.cmake:59]: I don't know if you users need more of an explanation than this:

```suggestion
    MESSAGE(FATAL_ERROR "You have requested C++17 support for NVCC ${KOKKOS_CXX_COMPILER_VERSION}, but you enabled relaxed constexpr. Please reduce the C++ standard to 14 or re-configure with -DKokkos_ENABLE_CUDA_CONSTEXPR=OFF")
```

File: cmake/kokkos_corner_cases.cmake (hunk 1 above)
Comment (dalg24) [reply cmake/kokkos_corner_cases.cmake:58]: You are entirely right

**Pull request closed without merging**